Lib/_collections_abc.py

```diff
@@ -409,6 +413,80 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _CallableGenericAlias(GenericAlias):
+    """ Internal class specifically for consistency between the ``__args__`` of
+    ``collections.abc.Callable``'s and ``typing.Callable``'s ``GenericAlias``.
+    """
+    __slots__ = ()
+    def __new__(cls, origin, args):
+        try:
+            return cls.__getitem_type(origin, args)
+        except TypeError:
+            # Fail-safe: most builtin generic collections don't validate the
+            # arguments passed to types.GenericAlias anyways.
+            # This is also because a subclass of the typing.Callable generic
+            # will have an __mro__ (<class 'MyCallable'>,
+            # <class 'collections.abc.Callable'>, <class 'typing.Generic'>,
+            # <class 'object'>). Subclasses wil use the __class_getitem__ of
+            # collections.abc.Callable before typing.Generic. As a result,
+            # we need to fall back on this to allow things like::
+            #
+            # T = TypeVar('T')
+            # class C1(typing.Callable[[T], T]): ...
+            # C1[int]
+            #
+            # Otherwise that will raise a TypeError.
```

Comment:
More typical examples are just `Callable[int]` and `Callable[int, int]`. (I had to think about why you chose this particular example (which isn't even legal according to PEP 484). Honestly do we need this try/except at all?

Reply:
Honestly, I wanted to remove the entire try except block and yes the example is admittedly contrived. The problem is with the try except removed, the following test in test_typing fails:
```python
    def test_extended_generic_rules_subclassing(self):
        class T1(Tuple[T, KT]): ...
        class T2(Tuple[T, ...]): ...
        class C1(Callable[[T], T]): ...
        class C2(Callable[..., int]):
            def __call__(self):
                return None

        [...]

        self.assertEqual(repr(C1[int]).split('.')[-1], 'C1[int]')
        [...]
```
As you can see, that typing test doesn't conform to PEP484 either. Should I instead change the test?

Reply:
> Honestly, I wanted to remove the entire try except block and yes the example is admittedly contrived. The problem is with the try except removed, the following test in test_typing fails:
> 
> ```python
>     def test_extended_generic_rules_subclassing(self):
>         class T1(Tuple[T, KT]): ...
>         class T2(Tuple[T, ...]): ...
>         class C1(Callable[[T], T]): ...
>         class C2(Callable[..., int]):
>             def __call__(self):
>                 return None
> 
>         [...]
> 
>         self.assertEqual(repr(C1[int]).split('.')[-1], 'C1[int]')
>         [...]
> ```
> 
> As you can see, that typing test doesn't conform to PEP484 either. Should I instead change the test?

Oh, that's unfortunate. Yes, let's delete those two classes and their uses from that test.